lib/keyboard.js

```diff
@@ -0,0 +1,134 @@
+(function(T) {
+    "use strict";
+
+    if (T.envtype !== "browser") {
+        return;
+    }
+
+    var fn = T.fn;
+    var instance = null;
+
+    function KeyboardListener(_args) {
+        if (instance) {
+            return instance;
+        }
+        instance = this;
+
+        T.Object.call(this, 1, _args);
+
+        fn.fixKR(this);
+    }
+    fn.extend(KeyboardListener);
+
+    var keyDown  = {};
+    var shiftKey = false;
+    var ctrlKey  = false;
+    var altKey   = false;
+
+    var onkeydown = function(e) {
+        var _ = instance._;
+        var cell = instance.cells[0];
+        var value = e.keyCode * _.mul + _.add;
+
+        for (var i = 0, imax = cell.length; i < imax; ++i) {
+            cell[i] = value;
+        }
+        shiftKey = e.shiftKey;
+        ctrlKey  = e.ctrlKey;
+        altKey   = e.altKey;
+
+        if (!keyDown[e.keyCode]) {
+            keyDown[e.keyCode] = true;
+            instance._.emit("keydown", e);
+        }
+    };
+
+    var onkeyup = function(e) {
+        delete keyDown[e.keyCode];
+        instance._.emit("keyup", e);
+    };
+
+    var $ = KeyboardListener.prototype;
```

Comment:
Using $ as a variable is going to confuse people who are used to using jQuery and associate $ with the jQuery object. Is there a reason not to use a more descriptive name like `keyListenerProto`?